azurerm/internal/provider/features.go

```diff
@@ -26,6 +26,20 @@ func schemaFeatures() *schema.Schema {
 						},
 					},
 				},
+
+				"virtual_machine_scale_set": {
+					Type:     schema.TypeList,
+					Optional: true,
+					MaxItems: 1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"roll_instances_when_required": {
+								Type:     schema.TypeBool,
+								Required: true,
```

Comment:
Should this be optional for when more options go in, or is the intention to require all provider features to be explicitly states in every provider block?

Reply:
that makes sense once we've added a secondary field here (which is why the parsing logic below handles this) but since there's only a single element it didn't feel worth making this field optional for the moment (since otherwise you may as well omit the `virtual_machine_scale_set` block?)